examples/lang/erc721/src/lib.rs

```diff
@@ -0,0 +1,233 @@
+// Copyright 2018-2019 Parity Technologies (UK) Ltd.
+// This file is part of ink!.
+//
+// ink! is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// ink! is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with ink!.  If not, see <http://www.gnu.org/licenses/>.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use ink_core::{
+    env::DefaultSrmlTypes,
+    memory::format,
+    storage,
+    storage::Flush,
+};
+use ink_lang::contract;
+use ink_model::EnvHandler;
+use scale::{Encode, Decode};
+
+pub type Result<T, E> = core::result::Result<T, E>;
+pub type TokenId = u32;
+
+#[derive(Encode, Decode)]
+#[cfg_attr(feature = "ink-generate-abi", derive(Metadata))]
+struct Counter {
+    value: storage::Value<u32>,
+}
+
+impl Counter {
+    pub fn increase(&mut self) {
+        self.value += 1;
+    }
+
+    pub fn decrease(&mut self) {
+        self.value -= 1;
+    }
+
+    pub fn get(&self) -> u32 {
+        *self.value
+    }
+}
+
+impl Flush for Counter {
+	fn flush(&mut self) {}
```

Comment:
you need to actually flush any storage entity in the `Flush` implementation.
so `storage::Value<u32>` should be flushed for a correct implementation.

Reply:
I did a few changes as you suggested. Do I still need to implement the flush if it is not anymore a `storage::Value<u32>`?
Flush is implemented at `storage::Value`